### ISSUE TITLE
Analyzers API1000 and API1001 should trigger on conditional returns

### ISSUE THREAD

**Issue by eatdrinksleepcode** [dotnet/aspnetcore#33105]
### Describe the solution you'd like
Analyzers API1000 and API1001 should trigger on returns (or [expression-bodied methods](#33091)) with conditional expressions.

### Additional context

```csharp
public ActionResult<object> OopsUndocumentedIf(int id)
{
    if (id == 0)
    {
        return NotFound(); // API1000
    }

    return Ok(new object());
}

public ActionResult<object> OopsUndocumentedConditional(int id)
{
    return id == 0 ? NotFound() : Ok(new object()); // No API1000
}
```

**Comment by pranavkm** [comment]
@eatdrinksleepcode I think rewriting these analyzers to use operation blocks should simplify authoring it and address a lot of these control-flow issues. Is that something you might be interested in trying to pick up?

**Comment by eatdrinksleepcode** [comment]
Agreed, using operations makes more sense with these semantic rules. I can look into that this weekend. Should I open a separate issue?

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @eatdrinksleepcode .
Let's use this issue to track it. Send out the PR and we'll look at it.

**Comment by singh733** [comment]
@pranavkm I have gone through the code and below is the proposed solution. Please see the TODO Comment :

![image](https://user-images.githubusercontent.com/83835739/122703894-2c84c180-d270-11eb-91f6-fd544b48cc8c.png)

**Comment by Youssef1313** [comment]
> @pranavkm I have gone through the code and below is the proposed solution. Please see the TODO Comment :
> 
> ![image](https://user-images.githubusercontent.com/83835739/122703894-2c84c180-d270-11eb-91f6-fd544b48cc8c.png)

It's better to completely rewrite using IOperation rather than using syntax analysis.

**Comment by Youssef1313** [comment]
Looks like this was already rewritten in https://github.com/dotnet/aspnetcore/pull/34020.

This will need a test to confirm whether this specific issue was fixed with the rewrite or not.

**Comment by eatdrinksleepcode** [comment]
@Youssef1313 the code was rewritten to use operations instead of syntax, but conditional returns were not handled as far as I can see. The new code is still examining the return for method calls and object creation, not for conditionals.

**Comment by pranavkm** [comment]
Yup - it was straight rewrite without adding new features. If you'd like to extend it further, we'd be happy to accept a PR for it.

**Comment by Youssef1313** [comment]
This analyzer was deprecated in .NET 10 and removed in .NET 11. See https://learn.microsoft.com/en-us/aspnet/core/breaking-changes/10/openapi-analyzers-deprecated

Related to https://github.com/dotnet/aspnetcore/issues/68087

cc @danroth27
